app/models/user_project_preference.rb

```diff
@@ -1,6 +1,9 @@
 class UserProjectPreference < ActiveRecord::Base
   include Preferences
 
+  class Unauthorized < StandardError; end
```

Comment:
This one should move the the controller layer in [api_errors.rb](https://github.com/zooniverse/Panoptes/blob/master/app/controllers/concerns/api_errors.rb), in the controller you can then just raise the api error itself.